lib/Doctrine/ORM/Proxy/ProxyFactory.php

```diff
@@ -71,10 +76,11 @@ class ProxyFactory extends AbstractProxyFactory
      * @param string                 $proxyNs      The namespace to use for the proxy classes.
      * @param boolean|int            $autoGenerate The strategy for automatically generating proxy classes. Possible
      *                                             values are constants of Doctrine\Common\Proxy\AbstractProxyFactory.
+     * @param int                    $umask        The umask that controls file permissions.
      */
-    public function __construct(EntityManagerInterface $em, $proxyDir, $proxyNs, $autoGenerate = AbstractProxyFactory::AUTOGENERATE_NEVER)
+    public function __construct(EntityManagerInterface $em, $proxyDir, $proxyNs, $autoGenerate = AbstractProxyFactory::AUTOGENERATE_NEVER, $umask = 0002)
     {
-        $proxyGenerator = new ProxyGenerator($proxyDir, $proxyNs);
+        $proxyGenerator = new ProxyGenerator($proxyDir, $proxyNs, $umask);
```

Comment:
This is a BC break and requires a version of doctrine/common with this patch included: https://github.com/doctrine/common/pull/711

Reply:
Technically not on call-side, it's a constructor - parameter has default value and constructors don't follow LSP.